src/language/dictionary/JMDict/Spelling.java

```diff
@@ -14,13 +13,15 @@
     private boolean isKanji;
     private String spelling;
     private Spelling[] dependencies;
+    private List<Spelling> readings;
```

Comment:
I should note that there are hundreds of thousands of instances of 'Spelling' objects when Spark Reader starts up (I just checked, it's 384 842). Luckily, a modern computer has enough RAM to hold the entirety of the Japanese language.

That being said, every byte added here adds several hundred kilobytes to the RAM usage, so I prefer to keep redundancy down unless it can help to improve performance or seriously improves readability. 

And yet, though it can be calculated on demand, I can see this to be a pretty useful property to store as well. I'm conflicted on whether I should keep this here or not, as I'm aware there's other parts of the code probably wasting far more memory, this one's just more obvious.

And perhaps it's because I just replaced that inefficient messy regex hell with a brand new clean and performance-minded solution, but I think I'll be taking this out and computing it on demand (as is planned with relevant tags and senses), instead of calculating and storing this for every spelling in the language on boot.